junit_tests/DateTimeTest.java

```diff
@@ -0,0 +1,707 @@
+            Int.getInt(2021),
+            Int.getInt(9),
+            Int.getInt(14),
+            Int.getInt(13),
+            Int.getInt(43),
+            Int.getInt(11),
+            Int.getInt(432523)
+        }, Collections.emptyMap());
+    }
+
+    @Test
+    public void testToday() {
+        DateTime.today(); // TODO: Mock?
+    }
+
+    @Test
+    public void testDate() {
+        //dateTime.date(); //TODO: Wait for Date to fix.
+    }
+
+    @Test
+    public void testWeekday() {
+        for (int i = 0; i < 7; i++) {
+            dateTime = new DateTime(new org.python.Object[]{
+                Int.getInt(2021),
+                Int.getInt(9),
+                Int.getInt(13 + i)
+            }, Collections.emptyMap());
+
+            assertEquals(Int.getInt(i), dateTime.weekday());
+        }
+    }
+
+    @Test
+    public void testStr() {
+        assertEquals(new org.python.types.Str("2021-09-14 13:43:11.432523"), dateTime.__str__());
+    }
+
+    @Test
+    public void testStrLowDateValues() {
+        dateTime = new DateTime(new org.python.Object[]{
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1)
+        }, Collections.emptyMap());
+
+        assertEquals(new org.python.types.Str("0001-01-01 01:01:01.000001"), dateTime.__str__());
+    }
+
+    @Test
+    public void testConstructorInvalidYear() {
+        String expectedMessage = "year 0 is out of range";
+        org.python.Object[] args = {
+            Int.getInt(0),
+            Int.getInt(1),
+            Int.getInt(1)
+        };
+        Exception exception = assertThrows(ValueError.class, () -> {
+            new DateTime(args, Collections.emptyMap());
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+
+        expectedMessage = "year 10000 is out of range";
+        args[0] = Int.getInt(10000);
+        exception = assertThrows(ValueError.class, () -> {
+            new DateTime(args, Collections.emptyMap());
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    public void testConstructorInvalidMonth() {
+        String expectedMessage = "month must be in 1..12";
+        org.python.Object[] args = {
+            Int.getInt(1),
+            Int.getInt(0),
+            Int.getInt(1)
+        };
+        Exception exception = assertThrows(ValueError.class, () -> {
+            new DateTime(args, Collections.emptyMap());
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+
+        args[1] = Int.getInt(13);
+        exception = assertThrows(ValueError.class, () -> {
+            new DateTime(args, Collections.emptyMap());
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    public void testConstructorInvalidDay() {
+        // TODO: Check for February and stuff.
```

Comment:
Remaining TODO, you should either finish it or remove it

Reply:
Removed it in 946bb34, because not necessary.